SingularityBase/src/main/java/com/hubspot/mesos/MesosUtils.java

```diff
@@ -183,33 +215,65 @@ public static Resource getPortsResource(int numPorts, List<Resource> resources,
         .build();
   }
 
-  private static Resource newScalar(String name, double value) {
-    return Resource.newBuilder().setName(name).setType(Value.Type.SCALAR).setScalar(Value.Scalar.newBuilder().setValue(value).build()).build();
+  private static Resource newScalar(String name, double value, Optional<String> role) {
+    Resource.Builder builder = Resource.newBuilder().setName(name).setType(Value.Type.SCALAR).setScalar(Value.Scalar.newBuilder().setValue(value).build());
+    if (role.isPresent()) {
+      builder.setRole(role.get());
+    }
+
+    return builder.build();
   }
 
   private static Resource newRange(String name, long begin, long end) {
     return Resource.newBuilder().setName(name).setType(Type.RANGES).setRanges(Ranges.newBuilder().addRange(Range.newBuilder().setBegin(begin).setEnd(end).build()).build()).build();
   }
 
+  public static String getRolesInfo(Offer offer) {
```

Comment:
it looks like you're just using this in logging statements -- I think the `toString()` of `getRoles(offer)` should suffice